code/modules/power/singularity/emitter.dm

```diff
@@ -43,8 +48,15 @@
 
 /obj/machinery/power/emitter/Initialize()
 	. = ..()
-	if(state == 2 && anchored)
+	if(state == STATE_WELDED && anchored)
 		connect_to_network()
+	if(emagged) //So they can start emagged, if for some ungodly reason you want to torment the engineers.
+		emag_act()

```

Comment:
This doesn't actually work b/c there's an `if(!emagged)` guarding the logic in `emag_act()`

Reply:
we're setting emagged = TRUE on line 59, that shouuuuld mean emag_act() works? will test

Reply:
can confirm it does work! this is specifically for emitters that start emagged (e.g. for asshole PoIs - see 56-59 ominous emitter as a template for those), rather than emitters that get emagged in-round (which also works, but because those call emag_act() directly rather than checking if they're emagged on initialize)

Reply:
Did you check that burst count and firerate were updated? Line 59 is in a class definition, meaning an instance of that class will set variables to their proscribed values (`emagged = TRUE`) _before_ called `Initialize()`.
`Initialize()` will correctly call `emag_act()`, but `emag_act()` will return immediately because of the `if(!emagged)` check on line 267 guarding all of the logic.

Reply:
@Varlaisvea Still waiting for a response.

Reply:
@Varlaisvea ping